test/acceptance/features/collection/step_definitions/collection.js

```
@@ -53,50 +71,43 @@ defineSupportCode(({ When, Then }) => {
   })
 
   Then(/^I can view the (.+) in the collection$/, async function (entityType, dataTable) {
-    const entityHeaderName = get(this.state, `${camelCase(entityType)}.header`)
+    const entityHeading = get(this.state, `${camelCase(entityType)}.heading`)
 
     await Collection
       .section.firstCollectionItem
       .waitForElementPresent('@header')
-      .assert.containsText('@header', entityHeaderName)
+      .assert.containsText('@header', entityHeading)
 
-    for (let row of dataTable.hashes()) {
-      const element = Collection.getSelectorForMetaListItem(row.label)
-      const expectedText = get(this.state, row.statePath)
+    for (const row of dataTable.hashes()) {
+      const metaListValueElement = Collection.getSelectorForMetaListItemValue(row.text)
+      const expectedMetaListText = get(this.state, row.expect)
 
-      if (expectedText) {
+      if (expectedMetaListText) {
         await Collection
           .section.firstCollectionItem
           .api.useXpath()
-          .assert.containsText(element.selector, expectedText)
+          .assert.containsText(metaListValueElement.selector, expectedMetaListText)
           .useCss()
-      } else {
-        if (row.statePath) {
-          Collection
-            .assert.fail(`Invalid statePath for ${row.label}`)
-        }
+      }
+    }
+  })
 
+  Then(/^the (.+) has badges$/, async function (entityType, dataTable) {
+    for (const row of dataTable.hashes()) {
+      const badgeValueElement = Collection.getSelectorForBadgeWithText(row.text)
+      const expectedBadgeValue = get(this.state, row.expect)
+
+      if (expectedBadgeValue) {
```

We can remove this `if` and let it blow up.